File: lista de exercicios/questão 5/questao.go
package main

import "fmt"

type Animal interface {
    Som()
}

type Cachorro struct{}

func (Cachorro) Som() {
    fmt.Println("O cachorro faz: Au Au")
}

type Gato struct{}

func (Gato) Som() {
    fmt.Println("O gato faz: Miau")
}

func EmitirSons(animais []Animal) {
    for _, animal := range animais {
        animal.Som()
    }
}

func main() {
    animais := []Animal{Cachorro{}, Gato{}}
    EmitirSons(animais)
}

File: lista de exercicios/questão 5/questao.py
class Animal:
    def som(self):
        print("Som de animal")

class Cachorro(Animal):
    def som(self):
        print("O cachorro faz: Au Au")

class Gato(Animal):
    def som(self):
        print("O gato faz: Miau")

def emitir_sons(animais):
    for animal in animais:
        animal.som()

animais = [Cachorro(), Gato()]
emitir_sons(animais)
